src/main/java/com/commercetools/sync/products/service/impl/ProductReferenceTransformServiceImpl.java

```diff
@@ -0,0 +1,496 @@
+package com.commercetools.sync.products.service.impl;
+
+import static com.commercetools.sync.commons.utils.ResourceIdentifierUtils.REFERENCE_ID_FIELD;
+import static com.commercetools.sync.commons.utils.ResourceIdentifierUtils.REFERENCE_TYPE_ID_FIELD;
+import static com.commercetools.sync.products.utils.ProductReferenceResolutionUtils.mapToProductDrafts;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import com.commercetools.sync.commons.models.GraphQlQueryResources;
+import com.commercetools.sync.commons.models.ResourceIdsGraphQlRequest;
+import com.commercetools.sync.commons.utils.ChunkUtils;
+import com.commercetools.sync.customobjects.helpers.CustomObjectCompositeIdentifier;
+import com.commercetools.sync.products.service.ProductReferenceTransformService;
+import com.commercetools.sync.services.impl.BaseTransformServiceImpl;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.models.Asset;
+import io.sphere.sdk.models.Reference;
+import io.sphere.sdk.products.AttributeContainer;
+import io.sphere.sdk.products.Price;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductDraft;
+import io.sphere.sdk.products.ProductLike;
+import io.sphere.sdk.products.ProductVariant;
+import io.sphere.sdk.products.attributes.Attribute;
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.types.CustomFields;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProductReferenceTransformServiceImpl extends BaseTransformServiceImpl
+    implements ProductReferenceTransformService {
+
+  private static final Logger LOGGER =
```

Comment:
We don't have a logger in the java-sync 🤔

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/697/commits/c15173a3fe3479816c7abb1d34ecc09a498b8f0e